### PR TITLE
feat(dashboards): Widget Viewer reuse table results for table widgets

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Updates the Widget Viewer to reuse table results from widget cards on the initial render to reduce repeat requests on the same data.

### CODE REVIEW

File: tests/js/spec/components/modals/widgetViewerModal.spec.tsx [hunk 1]
```diff
@@ -763,5 +769,87 @@ describe('Modals -> WidgetViewerModal', function () {
...
+    };
+    const mockWidget = {
+      title: 'Test Widget',
+      displayType: DisplayType.TABLE,
+      interval: '5m',
+      queries: [mockQuery],
+      WidgetType: WidgetType.DISCOVER,
+    };
+    beforeEach(function () {
+      eventsv2Mock = MockApiClient.addMockResponse({
```
Comment (priscilawebdev) [inline tests/js/spec/components/modals/widgetViewerModal.spec.tsx:811]: yesterday I got this [feedback](https://github.com/getsentry/sentry/pull/33302#discussion_r845160077) in my PR and was wondering if we should do the same here

**Comment by priscilawebdev** [comment]
when testing locally I noticed that the same widget displays less number of rows. Please see.

**Before:**
image


**After:**
image

is this expected?

**Comment by edwardgou-sentry** [comment]
> when testing locally I noticed that the same widget displays less number of rows. Please see.
> 
> **Before:** image
> 
> **After:** image
> 
> is this expected?

Hey @priscilawebdev, hmm that's not the expected result. I just tested myself as well but wasn't able to reproduce (I also used the same dashboard and widget). I noticed data in the chart matches the table in the second picture so it might be correct. Do you know if the project filters might be different between the two pictures?

**Review by narsaynorath — APPROVED** [review]

**Comment by priscilawebdev** [comment]
just double-checked, making sure no project is selected and the table seems to show the same data now, only the total differs... do you have an idea why?

**Before:**

image

**After:**
image

**Comment by edwardgou-sentry** [comment]
Hey @priscilawebdev , hmm not sure why it's still displaying different values in total events for you. Just tested against the branch and master, both are returning the same values for me.

Are you able to inspect the `events-meta` request on your browser? That might show if there are any differences in the request payload. The only other thing I can think of that might affect the results is if you are using two different accounts.

**Comment by priscilawebdev** [comment]
@edwardgou-sentry oh yes I was using different accounts. It might be it.. thanks for checking that 🙂

**Pull request merged**